python/ray/experimental/state/common.py

```diff
@@ -10,6 +10,18 @@
 
 DEFAULT_RPC_TIMEOUT = 30
 DEFAULT_LIMIT = 1000
+# It is used to identify exceptions in the middle of streaming response.
+# Streaming response sends the http header "before" it writes data
+# to the stream. However, if the operation is failed in the middle of
+# streaming process (e.g., unexpected node failure), streaming can fail
+# after the header is sent (which means we cannot re-sent the header).
+# In this case, this error hash code is used to identify the error.
+# Protocol is as follow:
+# - Streamed data will start from this hash value if there's an error.
+# - The data will look like f"{ERROR_HASH_CODE}::{error_msg}"
+# - When the client reads the data, if the data starts from the hash,
+#   it raises an exception.
+ERROR_HASH_CODE = "8b74caa46a390f1aa810356e70c9ced7"
```

Comment:
If there are other better ways to catch exceptions in the middle of streaming, please lmk... I feel this way is a bit hacky (without this, it will raise a random exception that's not actionable at all)

Reply:
huh i see. can you always use the first byte to indicate if this is a normal log or an exception ?

Reply:
Good idea. Addressed